data-prepper-api/src/main/java/org/opensearch/dataprepper/model/event/DataType.java

```diff
@@ -75,4 +75,22 @@ public String getTypeName() {
     static DataType fromTypeName(final String option) {
         return TYPES_MAP.get(option);
     }
+
+    public static boolean isSameType(final Object object, final String option) {
+        DataType type = fromTypeName(option);
+        if (type == null)
```

Comment:
We should have a check for object is null.

`/myfield typeof null`

Reply:
I do not think that's necessary. We already support `/myfield == null`. I do not think `typeof null` makes sense.